service/qos/qos_common.hh

```diff
@@ -36,8 +36,12 @@ namespace qos {
  *  a service level.
  */
 struct service_level_options {
+    static constexpr lowres_clock::duration delete_marker = lowres_clock::duration::max();
+
```

Comment:
How is delete_marker different from timeout == nullopt, below?

Reply:
If you need three options, how about variant<unspecified, deleted, lowres_clock::duration>?

Reply:
Well, now I don't actually need this marker at all, the code can be simplified even further. Originally, the purpose was to distinguish between setting `read_timeout = null and write_timeout = 5s` and `write_timeout = 5s`. The former would actively delete the per-service-level timeout value, the latter would simply not touch it and only set the write timeout. Since I kept all the timeout values in a single struct, I needed to represent both "please leave untouched" and "please delete" somehow. Now that there's only 1 timeout value, I no longer need this dance. Perhaps it will be useful to introduce it later when more service level options are added, but there's no point in complicating the code now.

Reply:
Hm, but on the other hand I **do** plan to add more per-service-level options soon. So perhaps I'll indeed make it an explicit variant in v3.

Reply:
Yes, there's a difference between a mutation to service level options (which can delete a setting) and s value of service level options (which can represent a setting that was deleted or never set, but not the deletion itself).